spec/lib/twingly/http_spec.rb

```diff
@@ -533,6 +535,66 @@ class CustomError < StandardError; end
     end
   end
 
+  RSpec.shared_examples "verifies proxy functionality" do |http_method|
+    context "when a proxy is provided" do
+      let(:proxy_pid_and_url) { HttpTestServer.spawn("proxy_server") }
+      let(:proxy_pid) { proxy_pid_and_url[0] }
+      let(:proxy_url) { proxy_pid_and_url[1] }
+
+      let(:target_pid_and_url) { HttpTestServer.spawn("echoed_headers_in_body") }
+      let(:target_pid) { target_pid_and_url[0] }
+      let(:target_url) { target_pid_and_url[1] }
+
+      let(:client) do
+        described_class.new(
+          base_user_agent: "Test Agent",
+          proxy: proxy_url
+        )
+      end
+
+      after do
+        HttpTestServer.stop(proxy_pid)
+        HttpTestServer.stop(target_pid)
+      end
+
+      it "routes requests through the proxy", vcr: false do # rubocop:disable RSpec/ExampleLength
+        with_real_http_connections do
+          response = case http_method
+                     when :get
+                       client.get(target_url)
+                     when :post
+                       client.post(target_url, body: "test")
+                     when :put
+                       client.put(target_url, body: "test")
+                     when :patch
+                       client.patch(target_url, body: "test")
+                     when :delete
+                       client.delete(target_url)
+                     end
+
+          expect(response.body).to include("HTTP_X_PROXIED_BY")
+        end
+      end
+    end
+  end
```

Comment:
It's possible to clean this up a bit and get rid of the case-when statement and the `http_method` argument, by using some already existing `let` statements from further up in the file:

```suggestion
  RSpec.shared_examples "verifies proxy functionality" do
    context "when a proxy is provided" do
      let(:proxy_pid_and_url) { HttpTestServer.spawn("proxy_server") }
      let(:proxy_pid) { proxy_pid_and_url[0] }
      let(:proxy_url) { proxy_pid_and_url[1] }

      let(:target_pid_and_url) { HttpTestServer.spawn("echoed_headers_in_body") }
      let(:target_pid) { target_pid_and_url[0] }
      let(:target_url) { target_pid_and_url[1] }

      let(:client) do
        described_class.new(
          base_user_agent: base_user_agent,
          proxy: proxy_url
        )
      end

      let(:url) { target_url }

      after do
        HttpTestServer.stop(proxy_pid)
        HttpTestServer.stop(target_pid)
      end

      it "routes requests through the proxy", vcr: false do
        with_real_http_connections do
          expect(response.fetch(:body)).to include("HTTP_X_PROXIED_BY")
        end
      end
    end
  end
```

Reply:
Cleaned up in 7894f8d :)